### PR TITLE
Convert TrackFromPVSelector from legacy producer to global producer

### PR DESCRIPTION
**Pull request by knoepfel**
This was a nicely written module.  Straightforward to convert.

### CODE REVIEW

**Comment by knoepfel** [comment]
@Dr15Jones

File: Validation/RecoTau/plugins/TrackFromPVSelector.cc [hunk 1]
```diff
@@ -1,100 +1,78 @@
...
-    if ( fabs(TrackIt->dxy(PV.position())) < max_dxy_ &&
-         fabs(TrackIt->dz(PV.position()))  < max_dz_  ){
-      goodTracks -> push_back(*TrackIt) ;
+void TrackFromPVSelector::produce(edm::StreamID, edm::Event& iEvent, edm::EventSetup const&) const
+{
+  auto goodTracks = std::make_unique<std::vector<reco::Track>>();
+
+  edm::Handle<std::vector<reco::Vertex>> vertices;
+  edm::Handle<std::vector<reco::Track>> tracks;
+  if (iEvent.getByToken(v_recoVertexToken_, vertices) && !vertices->empty() &&
```
Comment (Dr15Jones) [inline Validation/RecoTau/plugins/TrackFromPVSelector.cc]: This is not the same logic. Previously if the data product was missing we'd throw an exception at the `VertexHandle->` or `TrackHandle->` call. Now you will put in an empty container. Please remove the check on return value of `getByToken`.

File: Validation/RecoTau/plugins/TrackFromPVSelector.cc [hunk 2]
```diff
@@ -1,100 +1,78 @@
...
+      iEvent.getByToken(v_recoTrackToken_, tracks) && !tracks->empty())
+    {
+      auto const& vtxPos = vertices->front().position();
+      std::copy_if(std::cbegin(*tracks), std::cend(*tracks), std::back_inserter(*goodTracks),
+                   [this, &vtxPos](auto const& track) {
+                     return fabs(track.dxy(vtxPos)) < max_dxy_ && fabs(track.dz(vtxPos)) < max_dz_;
```
Comment (Dr15Jones) [inline Validation/RecoTau/plugins/TrackFromPVSelector.cc]: Could you switch to using `std::abs` ?

File: Validation/RecoTau/plugins/TrackFromPVSelector.cc (hunk 1 above)
Comment (knoepfel) [inline Validation/RecoTau/plugins/TrackFromPVSelector.cc]: Ack.  Good catch.  Yes I'll make the change.

File: Validation/RecoTau/plugins/TrackFromPVSelector.cc (hunk 2 above)
Comment (knoepfel) [inline Validation/RecoTau/plugins/TrackFromPVSelector.cc]: Yep.

**Comment by knoepfel** [comment]
@Dr15Jones Logic error fixed and pushed.

**Comment by Dr15Jones** [comment]
please test

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**